django/core/management/commands/makemessages.py

```diff
@@ -443,9 +437,131 @@ def is_ignored(path, ignore_patterns):
+            file_group.append(translatable)
+
+        for locale_dir, files in file_groups.items():
+            self.process_locale_dir(locale_dir, files)
+
+    def process_locale_dir(self, locale_dir, files):
+        u"""
+        Extract translatable literals from :param files:, creating or updating
+        the POT file for a given :param locale_dir:.
+
+        Uses the xgettext GNU gettext utility.
+        """
+        build_files = []
+
+        for translatable in files:
+            if self.verbosity > 1:
+                self.stdout.write('processing file %s in %s\n' % (
+                    translatable.file, translatable.dirpath
+                ))
+
+            if self.domain not in ('djangojs', 'django'):
+                continue
+
+            build_file = self.build_file_class(self, self.domain, translatable)
+
+            try:
+                build_file.preprocess()
+            except UnicodeDecodeError as e:
+                self.stdout.write(
+                    'UnicodeDecodeError: skipped file %s in %s (reason: %s)' % (
+                        translatable.file, translatable.dirpath, e,
+                    )
+                )
+                continue
+
+            build_files.append(build_file)
+
+        if self.domain == 'djangojs':
+            is_templatized = build_file.is_templatized
+            args = [
+                'xgettext',
+                '-d', self.domain,
+                '--language=%s' % ('C' if is_templatized else 'JavaScript',),
+                '--keyword=gettext_noop',
+                '--keyword=gettext_lazy',
+                '--keyword=ngettext_lazy:1,2',
+                '--keyword=pgettext:1c,2',
+                '--keyword=npgettext:1c,2,3',
+                '--output=-',
+            ]
+        elif self.domain == 'django':
+            args = [
+                'xgettext',
+                '-d', self.domain,
+                '--language=Python',
+                '--keyword=gettext_noop',
+                '--keyword=gettext_lazy',
+                '--keyword=ngettext_lazy:1,2',
+                '--keyword=ugettext_noop',
+                '--keyword=ugettext_lazy',
+                '--keyword=ungettext_lazy:1,2',
+                '--keyword=pgettext:1c,2',
+                '--keyword=npgettext:1c,2,3',
+                '--keyword=pgettext_lazy:1c,2',
+                '--keyword=npgettext_lazy:1c,2,3',
+                '--output=-',
+            ]
+        else:
+            return
+
+        input_files = [bf.work_path for bf in build_files]
+
+        fp, input_files_list = tempfile.mkstemp()
+        f = open(input_files_list, 'w')
+        try:
+            f.write('\n'.join(input_files))
+        finally:
+            f.close()
+
+        args.extend(['--files-from', input_files_list])
+        args.extend(self.xgettext_options)
+
+        msgs, errors, status = gettext_popen_wrapper(args)
+
+        os.unlink(input_files_list)
```

Comment:
There are problems on Windows with this line "The process cannot access the file because it is being used by another process." I think we have had similar issues before -- maybe you can search for some history if it's unclear how to solve it.